src/sentry/search/events/builder.py

```diff
@@ -22,16 +24,19 @@ def __init__(
         orderby: Optional[List[str]] = None,
         auto_aggregations: bool = False,
         use_aggregate_conditions: bool = False,
-        limit: int = 50,
-        offset: Optional[int] = 0,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+        limitby: Optional[Tuple[str, int]] = None,
     ):
         super().__init__(dataset, params)
 
         # TODO: implement this in `resolve_select`
         self.auto_aggregations = auto_aggregations
 
-        self.limit = Limit(limit)
-        self.offset = Offset(0 if offset is None else offset)
+        self.limit = None if limit is None else Limit(limit)
```

Comment:
Should there be a default still if both limit and limitby aren't set? I feel like this might be a different behavior?

Reply:
Opting for a default of 50 with an explicit way of setting it to `None` if necessary.